src/pages/listing/BookingCard.tsx

```diff
@@ -62,6 +65,24 @@ const BookingCard = ({
       <Button className="w-100">Request to Book</Button>
     </Row>
   </Card>;
+
+  function isDayBlocked(day: moment.Moment) {
+    const quantity = totalQuantity || 1;
+    const utcDay = day.clone().utc().set('hours', 0);
+    const pickingEnd = focusedInput === 'endDate';
+    if (pickingEnd && startDate) {
+      const selectedStartDate = startDate.utc().set('hours', 0);
+      return (
+        utcDay.isBefore(selectedStartDate) ||
+        reservations.filter(({ startDate, endDate }: Reservation) => {
+          return selectedStartDate.isBefore(endDate) && utcDay.isAfter(startDate);
+        }).length >= quantity
+      );
+    }
+    return reservations.filter(({ startDate, endDate }: Reservation) => {
+      return utcDay.isBetween(startDate, endDate, undefined, pickingEnd ? '(]' : '[)');
+    }).length >= quantity;
+  }
```

Comment:
@jerminatorhits This is mostly copypasta from BookingRequestCard. Does it look like I broke anything important bringing it over?

Reply:
copypastas are the most delicious pastas

Reply:
lgtm! tbh it's been so long since I've touched this section I don't rly remember 😅